implementations/rust/ockam/ockam_command/src/identity/create.rs

```diff
@@ -44,7 +55,16 @@ async fn run_impl(
         options.state.vaults.default()?.config
     };
     let vault = vault_config.get().await?;
-    let identity = Identity::create(&ctx, &vault).await?;
+    let identity = if let Some(kid) = cmd.key_id {
```

Comment:
I'm not entirely sure that passing a key_id is only applicable for kms vaults. 

Suppose we have keys already on a non-kms vault  (like the existing file-based one).   Could we want to generate an identity using an existing key there instead of generating a new one?.    That looks like something that could be general and not specific to kms.   *But*  that brings the question of what the key_id parameter *is*,  to be general it shouldn't be something special about kms.

We could do something like:
create the vault
`$ ockam vault create v1 --kms`

import the kms key.  This would compute the `KeyId`  and add the mapping in our  storage from `KeyId -> $kms-key-id`,  and return the `KeyId`   
` $ ockam vault import-key  v1  $kms-key-id` 
      
Then the --key-id  we optionally pass when creating an identity is always a (ockam)  KeyId,   and is not specific to a specific kind of vault.

Reply:
I was not at all concerned with the command-line interface. In fact I closed the previous PR because of the number of changes on the CLI layer that happened in the meantime. What you find in here is the bare minimum to have some workable example and I absolutely agree that it can and should be reworked to fit better into the intended CLI.

Reply:
:+1:  lets ticket it up @adrianbenavides

Reply:
For reference: https://github.com/build-trust/codebase/issues/493